### PR TITLE
Release of past reservations implemented

### PR DESCRIPTION
**Pull request by SofiaZarzabal**
- Added button to release past reservations
- The method about this functionality is in the database class

**Folder Structure**
folderStructure

**Database debug**
- The first image contains three saved reservations of which two are past.
BD RESERVATION SAVE

- The second shows that these two reserves have been released.
BD RESERVATION RELEASE

**Demostration Pixel 3a**
In this I setted a current reservation and a past one.
GIF PR7 Pixel 3a

**Demostration Nexus S**
In this both reservations are old
GIF PR7 Nexus S

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java [hunk 1]
```diff
@@ -78,26 +79,44 @@ public void enableButtonEnd() {
...
             message = context.getString(R.string.toast_parking_space_reservation_save);
             showMessage(context, message);
             activity.finish();
         }
     }
 
+    @Override
+    public void showReleasedPastReservations(int amountReservations) {
+        Context context = getContext();
+        String message;
```
Comment (BQuinteros) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: this message variable is not necessary i think

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java [hunk 2]
```diff
@@ -34,9 +34,9 @@ public void onButtonParkingSpaceReservationPickerPressed(DatePickerDialog.OnDate
     @Override
     public void onButtonParkingSpaceReservationSavePressed() {
         model.completeReservationInfo(view.getParkingSpace(), view.getSecurityCode());
-        ReservationVerificationResult reservationVerificationResult = model.checkFields(model.getReservation());
+        ReservationVerificationResult reservationVerificationResult = model.checkFields();
         if (reservationVerificationResult == ReservationVerificationResult.SUCCESS) {
-            model.makeReservation();
+            model.getDatabase().addReservation(model.getReservation());
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java]: the model should not return the database. please call a method into the model to do this

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java [hunk 3]
```diff
@@ -71,4 +70,9 @@ public void onTimeSetPressed(int hour, int minute) {
             view.showDateAndTimeEnd(model.getReservation().getFormattedDate(model.getReservation().getDateEnd()), model.getReservation().getFormattedTime(model.getReservation().getTimeEnd()));
         }
     }
+
+    @Override
+    public void onButtonParkingSpaceReservationDeletePressed() {
+        view.showReleasedPastReservations(model.getDatabase().releasePastReservations());
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java]: same here, just call a methos like model.releasePastReservations()

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java [hunk 4]
```diff
@@ -78,26 +79,44 @@ public void enableButtonEnd() {
 
     @Override
     public int getParkingSpace() {
-        return Integer.parseInt(binding.inputParkingSpaceReservationPlace.getText().toString());
+        String parkingSpace = binding.inputParkingSpaceReservationPlace.getText().toString();
+        if (parkingSpace.isEmpty()) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: this logic should be in the presenter, not in the view

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java [hunk 5]
```diff
@@ -78,26 +79,44 @@ public void enableButtonEnd() {
...
+            return Constants.NUMBER_MINUS_ONE;
+        }
+        return Integer.parseInt(parkingSpace);
     }
 
     @Override
     public int getSecurityCode() {
-        return Integer.parseInt(binding.inputParkingSpaceReservationCode.getText().toString());
+        String securityCode = binding.inputParkingSpaceReservationCode.getText().toString();
+        if (securityCode.isEmpty()) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java (hunk 1 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: Changed!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java (hunk 3 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java (hunk 2 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java (hunk 4 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java (hunk 5 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java]: Done!

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**